source/Agents/GP/LGPAgent.hpp

```diff
@@ -1,181 +1,201 @@
-    int i = 0;
-
-#ifndef NDEBUG
-    std::cout << "=========================================" << std::endl;
-
-    Sensors::wallDistance(grid, *this, SensorDirection::LEFT);
-    Sensors::wallDistance(grid, *this, SensorDirection::RIGHT);
-    Sensors::wallDistance(grid, *this, SensorDirection::ABOVE);
-    Sensors::wallDistance(grid, *this, SensorDirection::BELOW);
-#endif
-
-    if (currentInstructionIndex != 0) {
-      resultsList[currentInstructionIndex - 1] = action_result;
-    } else {
-      resultsList[LISTSIZE - 1] = action_result;
-    }
-
-    while (i < LISTSIZE && action.empty()) {
-      if (std::find(actionsList.begin(), actionsList.end(),
-                    get<0>(instruction)) != actionsList.end()) {
-        action = get<0>(instruction);
-      } else if (std::find(sensorsNamesList.begin(), sensorsNamesList.end(),
-                           get<0>(instruction)) != sensorsNamesList.end()) {
-        // the instruction is in the sensor list (getLeft, getRight, getUp,
-        // getDown)
-        sensor = get<0>(instruction);
-
-        SensorDirection direction = Sensors::getSensorDirectionEnum(sensor);
-        int distance = Sensors::wallDistance(grid, *this, direction);
-        resultsList[currentInstructionIndex] = distance;
-      }
-
-      else {
-        // the instruction is an operation (lessthan, greaterthan, equals)
-        operation = get<0>(instruction);
-        if (operation == "lessthan") {
-          if (get<1>(instruction) < get<2>(instruction)) {
-            resultsList[currentInstructionIndex] = 1;
-            ++currentInstructionIndex;
-          } else {
-            resultsList[currentInstructionIndex] = 0;
-          }
-        } else if (operation == "greaterthan") {
-          if (get<1>(instruction) > get<2>(instruction)) {
-            resultsList[currentInstructionIndex] = 1;
-            ++currentInstructionIndex;
-          } else {
-            resultsList[currentInstructionIndex] = 0;
-          }
-        } else if (operation == "equals") {
-          if (get<1>(instruction) == get<2>(instruction)) {
-            resultsList[currentInstructionIndex] = 1;
-            ++currentInstructionIndex;
-          } else {
-            resultsList[currentInstructionIndex] = 0;
-          }
+#include "./GPAgent_.hpp"
+
+namespace cowboys
+{
+    const int LISTSIZE = 100;
+
+    class LGPAgent : public GPAgent_
+    {
+    protected:
+        // A dictionary of actions and a dictionary of sensors
+        // A sensor is a function that takes in a grid and returns a value (e.g. distance to nearest agent)
+
+        // For example group 1 has a function for the shortest path
+
+        std::vector<std::string> possibleInstructionsList = {};
+        std::vector<std::string> actionsList = {"up", "down", "left", "right"};
+        std::vector<std::string> operationsList = {"lessthan", "greaterthan", "equals"};
+        std::vector<int> resultsList = std::vector<int>(LISTSIZE);
```

Comment:
Is this just an arbitrary buffer for memory?

Reply:
I found the resultsList a little confusing throughout. Some comments here would help a lot.

Reply:
@FergusonAJ

Reply:
The resultsList is just a vector whose length matches that of the instructionsList. Each time an instruction in the instructionsList executes, whether it's a sensor, a comparison, or an action, it stores the result of that instruction into the corresponding index in the resultsList. That way a comparison instruction can look back at the resultsList and see the result of a previous action.